### PR TITLE
Targeting data for Contributions API

### PR DESCRIPTION
**Pull request by andre1050**
## What does this change?
Updates the Contributions API payload inside the SlotBodyEnd component to include the article related data that we'll need to implement targeting on the service.

## Why?
So the service has the data it needs to replicate the targeting logic in frontend.

### CODE REVIEW

File: fixtures/CAPI.ts [hunk 1]
```diff
@@ -8,6 +8,15 @@ export const CAPI: CAPIType = {
...
     webPublicationDateDisplay: 'Fri 02 Mar 2018 09.45 GMT',
+    pageType: {
+        hasShowcaseMainElement: false,
+        isFront: false,
+        isLiveblog: false,
+        isMinuteArticle: false,
+        isPaidContent: false,
+        isPreview: false,
+        isSensitive: false,
+    },
```
Comment (oliverlloyd) [inline fixtures/CAPI.ts]: I know that CAPI has this as a separate object but I can't see any good reason to do this so maybe we can flatten these properties when we bring them into DCR?

File: src/web/components/SlotBodyEnd.tsx [hunk 2]
```diff
@@ -6,24 +6,54 @@ const wrapperMargins = css`
...
+    // Putting together the request payload
+    const contributionsPayload = {
+        tracking: {
+            ophanPageId: window.guardian.config.ophan.pageViewId,
+            ophanComponentId: 'ACQUISITIONS_EPIC',
+            platformId: 'GUARDIAN_WEB',
+            campaignCode: 'gdnwb_copts_memco_remote_epic_test_api',
+            abTestName: 'remote_epic_test',
+            abTestVariant: 'api',
+            referrerUrl: window.location.href.split('?')[0],
```
Comment (oliverlloyd) [inline src/web/components/SlotBodyEnd.tsx]: Could this code ever end up being run on the server? If so, we might need to guard against window being undefined

File: src/web/components/SlotBodyEnd.tsx [hunk 3]
```diff
@@ -6,24 +6,54 @@ const wrapperMargins = css`
...
     };
+
+    const endpointUrl = 'https://contributions.guardianapis.com/epic';
     const { data, error } = useApi(endpointUrl, {
         method: 'POST',
         headers: {
             'Content-Type': 'application/json',
         },
-        body: JSON.stringify(trackingParams),
+        body: JSON.stringify(contributionsPayload),
```
Comment (oliverlloyd) [inline src/web/components/SlotBodyEnd.tsx]: This isn't one for this PR but just so I don't forget: do we need to consider escaping this payload?

File: fixtures/CAPI.ts (hunk 1 above)
Comment (andre1050) [reply fixtures/CAPI.ts]: As discussed irl, even though your suggestion would make sense for DCR, it's slightly outside of the scope of what this PR is supposed to do, so my preference is to use the `pageType` as is and consider a refactoring task in the DCR backlog.

File: src/web/components/SlotBodyEnd.tsx (hunk 2 above)
Comment (andre1050) [reply src/web/components/SlotBodyEnd.tsx]: Good question - nothing stopping a developer from dropping the component in a SSR component tree, even through they wouldn't get much value out out it. My preference is to ensure the component doesn't break if used server-side but without having to implement server-side analogous logic, which right now we don't need.
I've added the optional chaining operator to `window?.` so no exceptions in the component - only window specific data would be sent as `undefined` to the API, which feels ok considering we don't always have a server-side way of accessing that data (e.g. Ophan ID).

File: src/web/components/SlotBodyEnd.tsx (hunk 3 above)
Comment (andre1050) [reply src/web/components/SlotBodyEnd.tsx]: I believe `JSON.stringify()` will do everything we need.

**Pull request merged**